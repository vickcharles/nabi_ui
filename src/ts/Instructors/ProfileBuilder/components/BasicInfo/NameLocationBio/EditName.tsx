import * as React from 'react';

import Button from 'material-ui/Button';
import { FormControl, FormControlLabel } from 'material-ui/Form';
import Radio, { RadioGroup } from 'material-ui/Radio';
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from 'material-ui/Dialog';

interface EditNameProps {
  firstName: string;
  lastName: string;
  isFormDialogOpen: boolean;
  closeHandler: () => void;
}
const EditName: React.StatelessComponent<EditNameProps> = props => {
  const lastNameinitial = props.lastName.charAt(0);

  return (
    <div>
      <Dialog
        open={props.isFormDialogOpen}
        onClose={props.closeHandler}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Edit Name</DialogTitle>
        <DialogContent>
          <DialogContentText className="nabi-margin-bottom-small">
            Choose display name
          </DialogContentText>
          <FormControl component="fieldset" required={true}>
          <RadioGroup
            name="displayName"
            value=""
          >
            <FormControlLabel 
              className="nabi-margin-bottom-xsmall"
              value="" 
              control={<Radio />} 
              label={`Show my full name: ${props.firstName} ${props.lastName}`} 
            />
            <FormControlLabel 
              value="" 
              control={<Radio />} 
              label={`Show only my first name and last initial: ${props.firstName} ${lastNameinitial}.`}
            />
          </RadioGroup>
        </FormControl>
        </DialogContent>
        <DialogActions>
          <Button onClick={props.closeHandler} color="default">
            Close
          </Button>
          <Button variant="raised" onClick={props.closeHandler} color="primary">
            Save
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default EditName;
