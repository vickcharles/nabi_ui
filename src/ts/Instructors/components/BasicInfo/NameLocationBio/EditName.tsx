import * as React from 'react';

import Button from 'material-ui/Button';
import { FormControl, FormControlLabel } from 'material-ui/Form';
import Radio, { RadioGroup } from 'material-ui/Radio';
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from 'material-ui/Dialog';

/**
 * Props for EditName
 * @interface {EditNameProps}
 */
interface EditNameProps {
  displayName: string;
  firstName: string;
  lastName: string;
  isFormDialogOpen: boolean;
  closeHandler: () => void;
  handleChange: (event: React.FormEvent<{}>) => void;
  handleSubmit: (event: React.FormEvent<{}>) => void;
}

/**
 * Displays a dialog to change the user's displayName
 * @param {EditNameProps} props - The component's props
 */
const EditName: React.StatelessComponent<EditNameProps> = props => {
  const lastNameinitial = props.lastName.charAt(0);
  const fullName = `${props.firstName} ${props.lastName}`;
  const nameAndLasInitial = `${props.firstName} ${lastNameinitial}.`;

  return (
    <div>
      <Dialog
        open={props.isFormDialogOpen}
        onClose={props.closeHandler}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Edit Name</DialogTitle>
        <DialogContent>
          <DialogContentText className="nabi-margin-bottom-small">
            Choose display name
          </DialogContentText>
          <FormControl component="fieldset" required={true}>
          <RadioGroup
            name="displayName"
            value={props.displayName}
            onChange={props.handleChange}
          >
            <FormControlLabel 
              className="nabi-margin-bottom-xsmall"
              value={fullName}
              control={<Radio />} 
              label={`Show my full name: ${fullName}`} 
            />
            <FormControlLabel 
              value={nameAndLasInitial}
              control={<Radio />} 
              label={`Show only my first name and last initial: ${nameAndLasInitial}`}
            />
          </RadioGroup>
        </FormControl>
        </DialogContent>
        <DialogActions>
          <Button onClick={props.closeHandler} color="default">
            Close
          </Button>
          <Button variant="raised" onClick={props.handleSubmit} color="primary">
            Save
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default EditName;
