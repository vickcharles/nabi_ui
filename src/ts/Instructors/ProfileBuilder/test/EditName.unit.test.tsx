import * as React from 'react';
import EditName from '../components/BasicInfo/NameLocationBio/EditName';
import { mount, shallow } from 'enzyme';

describe('EditName', () => {
  let wrapper: any;
  const closeHandler: () => void = jest.fn();
  
  describe('Mount', () => {
    beforeEach(() => {
      wrapper = mount(
        <EditName
          firstName=""
          lastName=""
          isFormDialogOpen={false}
          closeHandler={closeHandler}
        />
      );
    });
  });

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <EditName
          firstName=""
          lastName=""
          isFormDialogOpen={false}
          closeHandler={closeHandler}
        />
      );
    });
    
    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
