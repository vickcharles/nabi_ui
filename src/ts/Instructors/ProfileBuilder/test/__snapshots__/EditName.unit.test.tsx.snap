// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditName Shallow Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditName
    closeHandler={[MockFunction]}
    firstName=""
    isFormDialogOpen={false}
    lastName=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        aria-labelledby="form-dialog-title"
        onClose={[MockFunction]}
        open={false}
      >
        <WithStyles(DialogTitle)
          id="form-dialog-title"
        >
          Edit Name
        </WithStyles(DialogTitle)>
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)
            className="nabi-margin-bottom-small"
          >
            Choose display name
          </WithStyles(DialogContentText)>
          <WithStyles(FormControl)
            component="fieldset"
            required={true}
          >
            <RadioGroup
              name="displayName"
              value=""
            >
              <WithStyles(FormControlLabel)
                className="nabi-margin-bottom-xsmall"
                control={<WithStyles(Radio) />}
                label="Show my full name:  "
                value=""
              />
              <WithStyles(FormControlLabel)
                control={<WithStyles(Radio) />}
                label="Show only my first name and last initial:  ."
                value=""
              />
            </RadioGroup>
          </WithStyles(FormControl)>
        </WithStyles(DialogContent)>
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="default"
            onClick={[MockFunction]}
          >
            Close
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[MockFunction]}
            variant="raised"
          >
            Save
          </WithStyles(Button)>
        </WithStyles(DialogActions)>
      </WithStyles(Dialog)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-labelledby": "form-dialog-title",
        "children": Array [
          <WithStyles(DialogTitle)
            id="form-dialog-title"
          >
            Edit Name
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)
              className="nabi-margin-bottom-small"
            >
              Choose display name
            </WithStyles(DialogContentText)>
            <WithStyles(FormControl)
              component="fieldset"
              required={true}
            >
              <RadioGroup
                name="displayName"
                value=""
              >
                <WithStyles(FormControlLabel)
                  className="nabi-margin-bottom-xsmall"
                  control={<WithStyles(Radio) />}
                  label="Show my full name:  "
                  value=""
                />
                <WithStyles(FormControlLabel)
                  control={<WithStyles(Radio) />}
                  label="Show only my first name and last initial:  ."
                  value=""
                />
              </RadioGroup>
            </WithStyles(FormControl)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="default"
              onClick={[MockFunction]}
            >
              Close
            </WithStyles(Button)>
            <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
              variant="raised"
            >
              Save
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "onClose": [MockFunction],
        "open": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Edit Name",
            "id": "form-dialog-title",
          },
          "ref": null,
          "rendered": "Edit Name",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(DialogContentText)
                className="nabi-margin-bottom-small"
              >
                Choose display name
              </WithStyles(DialogContentText)>,
              <WithStyles(FormControl)
                component="fieldset"
                required={true}
              >
                <RadioGroup
                  name="displayName"
                  value=""
                >
                  <WithStyles(FormControlLabel)
                    className="nabi-margin-bottom-xsmall"
                    control={<WithStyles(Radio) />}
                    label="Show my full name:  "
                    value=""
                  />
                  <WithStyles(FormControlLabel)
                    control={<WithStyles(Radio) />}
                    label="Show only my first name and last initial:  ."
                    value=""
                  />
                </RadioGroup>
              </WithStyles(FormControl)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Choose display name",
                "className": "nabi-margin-bottom-small",
              },
              "ref": null,
              "rendered": "Choose display name",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <RadioGroup
                  name="displayName"
                  value=""
                >
                  <WithStyles(FormControlLabel)
                    className="nabi-margin-bottom-xsmall"
                    control={<WithStyles(Radio) />}
                    label="Show my full name:  "
                    value=""
                  />
                  <WithStyles(FormControlLabel)
                    control={<WithStyles(Radio) />}
                    label="Show only my first name and last initial:  ."
                    value=""
                  />
                </RadioGroup>,
                "component": "fieldset",
                "required": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <WithStyles(FormControlLabel)
                      className="nabi-margin-bottom-xsmall"
                      control={<WithStyles(Radio) />}
                      label="Show my full name:  "
                      value=""
                    />,
                    <WithStyles(FormControlLabel)
                      control={<WithStyles(Radio) />}
                      label="Show only my first name and last initial:  ."
                      value=""
                    />,
                  ],
                  "name": "displayName",
                  "value": "",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "nabi-margin-bottom-xsmall",
                      "control": <WithStyles(Radio) />,
                      "label": "Show my full name:  ",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "control": <WithStyles(Radio) />,
                      "label": "Show only my first name and last initial:  .",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="default"
                onClick={[MockFunction]}
              >
                Close
              </WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[MockFunction]}
                variant="raised"
              >
                Save
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Close",
                "color": "default",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Close",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Save",
                "color": "primary",
                "onClick": [MockFunction],
                "variant": "raised",
              },
              "ref": null,
              "rendered": "Save",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          aria-labelledby="form-dialog-title"
          onClose={[MockFunction]}
          open={false}
        >
          <WithStyles(DialogTitle)
            id="form-dialog-title"
          >
            Edit Name
          </WithStyles(DialogTitle)>
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)
              className="nabi-margin-bottom-small"
            >
              Choose display name
            </WithStyles(DialogContentText)>
            <WithStyles(FormControl)
              component="fieldset"
              required={true}
            >
              <RadioGroup
                name="displayName"
                value=""
              >
                <WithStyles(FormControlLabel)
                  className="nabi-margin-bottom-xsmall"
                  control={<WithStyles(Radio) />}
                  label="Show my full name:  "
                  value=""
                />
                <WithStyles(FormControlLabel)
                  control={<WithStyles(Radio) />}
                  label="Show only my first name and last initial:  ."
                  value=""
                />
              </RadioGroup>
            </WithStyles(FormControl)>
          </WithStyles(DialogContent)>
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="default"
              onClick={[MockFunction]}
            >
              Close
            </WithStyles(Button)>
            <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
              variant="raised"
            >
              Save
            </WithStyles(Button)>
          </WithStyles(DialogActions)>
        </WithStyles(Dialog)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-labelledby": "form-dialog-title",
          "children": Array [
            <WithStyles(DialogTitle)
              id="form-dialog-title"
            >
              Edit Name
            </WithStyles(DialogTitle)>,
            <WithStyles(DialogContent)>
              <WithStyles(DialogContentText)
                className="nabi-margin-bottom-small"
              >
                Choose display name
              </WithStyles(DialogContentText)>
              <WithStyles(FormControl)
                component="fieldset"
                required={true}
              >
                <RadioGroup
                  name="displayName"
                  value=""
                >
                  <WithStyles(FormControlLabel)
                    className="nabi-margin-bottom-xsmall"
                    control={<WithStyles(Radio) />}
                    label="Show my full name:  "
                    value=""
                  />
                  <WithStyles(FormControlLabel)
                    control={<WithStyles(Radio) />}
                    label="Show only my first name and last initial:  ."
                    value=""
                  />
                </RadioGroup>
              </WithStyles(FormControl)>
            </WithStyles(DialogContent)>,
            <WithStyles(DialogActions)>
              <WithStyles(Button)
                color="default"
                onClick={[MockFunction]}
              >
                Close
              </WithStyles(Button)>
              <WithStyles(Button)
                color="primary"
                onClick={[MockFunction]}
                variant="raised"
              >
                Save
              </WithStyles(Button)>
            </WithStyles(DialogActions)>,
          ],
          "onClose": [MockFunction],
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Edit Name",
              "id": "form-dialog-title",
            },
            "ref": null,
            "rendered": "Edit Name",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(DialogContentText)
                  className="nabi-margin-bottom-small"
                >
                  Choose display name
                </WithStyles(DialogContentText)>,
                <WithStyles(FormControl)
                  component="fieldset"
                  required={true}
                >
                  <RadioGroup
                    name="displayName"
                    value=""
                  >
                    <WithStyles(FormControlLabel)
                      className="nabi-margin-bottom-xsmall"
                      control={<WithStyles(Radio) />}
                      label="Show my full name:  "
                      value=""
                    />
                    <WithStyles(FormControlLabel)
                      control={<WithStyles(Radio) />}
                      label="Show only my first name and last initial:  ."
                      value=""
                    />
                  </RadioGroup>
                </WithStyles(FormControl)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Choose display name",
                  "className": "nabi-margin-bottom-small",
                },
                "ref": null,
                "rendered": "Choose display name",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <RadioGroup
                    name="displayName"
                    value=""
                  >
                    <WithStyles(FormControlLabel)
                      className="nabi-margin-bottom-xsmall"
                      control={<WithStyles(Radio) />}
                      label="Show my full name:  "
                      value=""
                    />
                    <WithStyles(FormControlLabel)
                      control={<WithStyles(Radio) />}
                      label="Show only my first name and last initial:  ."
                      value=""
                    />
                  </RadioGroup>,
                  "component": "fieldset",
                  "required": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <WithStyles(FormControlLabel)
                        className="nabi-margin-bottom-xsmall"
                        control={<WithStyles(Radio) />}
                        label="Show my full name:  "
                        value=""
                      />,
                      <WithStyles(FormControlLabel)
                        control={<WithStyles(Radio) />}
                        label="Show only my first name and last initial:  ."
                        value=""
                      />,
                    ],
                    "name": "displayName",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "className": "nabi-margin-bottom-xsmall",
                        "control": <WithStyles(Radio) />,
                        "label": "Show my full name:  ",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "control": <WithStyles(Radio) />,
                        "label": "Show only my first name and last initial:  .",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Button)
                  color="default"
                  onClick={[MockFunction]}
                >
                  Close
                </WithStyles(Button)>,
                <WithStyles(Button)
                  color="primary"
                  onClick={[MockFunction]}
                  variant="raised"
                >
                  Save
                </WithStyles(Button)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Close",
                  "color": "default",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": "Close",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Save",
                  "color": "primary",
                  "onClick": [MockFunction],
                  "variant": "raised",
                },
                "ref": null,
                "rendered": "Save",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
